qiskit/aqua/algorithms/minimum_eigen_solvers/qaoa/qaoa.py

```diff
@@ -69,6 +70,7 @@ def __init__(self,
                  initial_state: Optional[InitialState] = None,
                  mixer: Union[OperatorBase, LegacyBaseOperator] = None,
                  initial_point: Optional[np.ndarray] = None,
+                 gradient: Optional[Union[GradientBase, Callable]] = None,
```

Comment:
Of course.
Would it make sense to update the type hints for gradient functions in the optimiser first? 
Currently they are given as
`            gradient_function (callable) : A function that
                computes the gradient of the objective function, or
                None if not available.`
We could then integrate just the same type hints here.

Reply:
Ok, we if we leave it for now in terms of the typehints but can it explain that this callable is given directly to the optimizer - at least I think it is, so at least people can look at the scipy optimizer docs to see more about this since really it needs to be conformant to that if its the same thing.